server/game/cards/23-AHaH/TorrhensSquare.js

```diff
@@ -10,12 +10,13 @@ class TorrhensSquare extends DrawCard {
             when: {
                 onCardOutOfShadows: event => event.card.controller === this.controller && event.card.isFaction('greyjoy')
             },
-            cost: ability.costs.kneelFactionCard(),
+            cost: ability.costs.kneelSelf(),
             message: {
-                format: '{player} kneels their faction card to discard the top {amount} cards from {opponents} deck',
+                format: '{player} kneels {this} to discard the top {amount} cards from {opponents} deck',
```

Comment:
Should just be able to use `{source}` here. Don't forget to remove `this` in args when you do!
```suggestion
                format: '{player} kneels {source} to discard the top {amount} cards from {opponents} deck',
```